zig/private/common/zig_build.bzl

```
@@ -125,9 +125,15 @@ def zig_build_impl(ctx, *, kind):
         static = ctx.actions.declare_file(ctx.label.name + extension)
         outputs.append(static)
         args.add(static, format = "-femit-bin=%s")
-        # TODO[AH] Support dynamic library output.
 
         files = depset([static])
+    elif kind == "zig_shared_library":
+        extension = ".dll" if zigtargetinfo.triple.os == "windows" else ".so"
+        dynamic = ctx.actions.declare_file(ctx.label.name + extension)
+        outputs.append(dynamic)
+        args.add(dynamic, format = "-femit-bin=%s")
```

This produces outputs following the target name directly, e.g. the following produces `library.so`.

```
$ cd e2e/workspace
$ bazel build //simple-shared-library:library
...
Target //simple-shared-library:library up-to-date:
  bazel-bin/simple-shared-library/library.so
...
```

On Linux and MacOS the linker typically expects a `lib` prefix, so in that case the rule should generate an output. I'm not a Windows expert, but, [to my understanding](https://learn.microsoft.com/en-us/cpp/build/walkthrough-creating-and-using-a-dynamic-link-library-cpp?view=msvc-170) Windows does not use the `lib` prefix.

The following change should achieve this:
```suggestion
        prefix = "" if zigtargetinfo.triple.os == "windows" else "lib"
        extension = ".dll" if zigtargetinfo.triple.os == "windows" else ".so"
        dynamic = ctx.actions.declare_file(prefix + ctx.label.name + extension)
        outputs.append(dynamic)
        args.add(dynamic, format = "-femit-bin=%s")
```